std/utf.d

```diff
@@ -1328,10 +1348,8 @@ if (isSomeChar!(ElementEncodingType!S))
  * Returns:
  *      decoded character
  */
-private dchar decodeImpl(bool canIndex, UseReplacementDchar useReplacementDchar = No.useReplacementDchar, S)(
-    auto ref S str, ref size_t index)
-if (
-    is(S : const char[]) || (isInputRange!S && is(Unqual!(ElementEncodingType!S) == char)))
+private auto decodeImpl(bool canIndex, DecodeType decodeType, S)( auto ref S str, ref size_t index)
```

Comment:
This is only doable via a Dscanner plugin. However there's very good news: http://forum.dlang.org/thread/abbprxuwgqlmuuwdfgmg@forum.dlang.org & https://github.com/dlang-community/D-Scanner/issues/421